extensions/radial_profile_spotfinder_threshold_ext.py

```diff
@@ -0,0 +1,186 @@
+    """
+
+    name = "radial_profile"
+
+    @staticmethod
+    def phil():
+        from libtbx.phil import parse
+
+        phil = parse(
+            """
+        n_sigma = 8
+          .type = int
+          .help = "Sigma multiplier for determining the threshold value"
+
+        blur = narrow wide
+          .type = choice
+          .help = "Optional preprocessing of the image by a convolution with"
+                  "a simple Gaussian kernel of size either 3×3 (narrow) or"
+                  "5×5 (wide). This may help to reduce noise peaks and to"
+                  "combine split spots."
+
+        n_bins = 100
+          .type = int
+          .help = "Number of 2θ bins in which to calculate background"
+        """
+        )
+        return phil
+
+    def __init__(self, params):
+        """
+        Initialise the algorithm.
+
+        :param params: The input parameters
+        """
+        self.params = params
+
+        # Set approximate Gaussian kernel for blurring
+        if self.params.spotfinder.threshold.radial_profile.blur == "narrow":
+            self.kernel = flex.double(
+                (0.0625, 0.125, 0.0625, 0.125, 0.25, 0.125, 0.0625, 0.125, 0.0625)
+            )
+            self.kernel.reshape(flex.grid((3, 3)))
+        elif self.params.spotfinder.threshold.radial_profile.blur == "narrow":
+            self.kernel = (
+                flex.double(
+                    (
+                        1,
+                        4,
+                        7,
+                        4,
+                        1,
+                        4,
+                        16,
+                        26,
+                        16,
+                        4,
+                        7,
+                        26,
+                        41,
+                        26,
+                        7,
+                        4,
+                        16,
+                        26,
+                        16,
+                        4,
+                        1,
+                        4,
+                        7,
+                        4,
+                        1,
+                    )
+                )
+                / 273
+            )
+            self.kernel.reshape(flex.grid((5, 5)))
+        else:
+            self.kernel = None
+
+    def compute_threshold(self, image, mask, **kwargs):
+        """
+        Compute the threshold.
+
+        :param image: The image to process
+        :param mask: The pixel mask on the image
+        :**kwargs: Arbitrary keyword arguments
+        :returns: A boolean mask showing foreground/background pixels
+        """
+
+        if self.kernel:
+            image = convolve(image, self.kernel)
+
+        try:
+            imageset = kwargs["imageset"]
+            i_panel = kwargs["i_panel"]
+        except KeyError:
+            raise ValueError("Missing required arguments")
+        region_of_interest = kwargs.get("region_of_interest")
```

Comment:
I think you can use the [keyword-only asterisk syntax](https://www.python.org/dev/peps/pep-3102/) to simplify this section:
```suggestion
    def compute_threshold(self, image, mask, *, imageset, i_panel, region_of_interest=None, **kwargs):
        """
        Compute the threshold.

        :param image: The image to process
        :param mask: The pixel mask on the image
        :**kwargs: Arbitrary keyword arguments
        :returns: A boolean mask showing foreground/background pixels
        """

        if self.kernel:
            image = convolve(image, self.kernel)
```